charts/argo-templates/templates/download-ftp-file.yaml

```diff
@@ -0,0 +1,49 @@
+{{- with .Values.downloadFTPFile }}
+{{- if .create }}
+apiVersion: argoproj.io/v1alpha1
+kind: WorkflowTemplate
+metadata:
+  name: {{ .name }}
+spec:
+  templates:
+    - name: main
+      inputs:
+        parameters:
+          - name: ftp-site
+          - name: ftp-path
+          - name: local-path
+          - name: pvc-name
+          - name: memory
+          - name: cpu
+      volumes:
+        - name: state
+          persistentVolumeClaim:
+            claimName: '{{ "{{inputs.parameters.pvc-name}}" }}'
+      {{- include "argo.retry" . | indent 6 }}
```

Comment:
Unclear to me if this should become parameterized on a per-template basis. Right now all of our retries are basically "hit it with a hammer until it works" situations, but it's plausible that we'll want more fine-grained control in the future.